integration-tests/src/test/java/oracle/kubernetes/operator/ITServerDiscovery.java

```diff
@@ -0,0 +1,541 @@
+    operator.scale(domainUid, clusterName, replicas);
+
+    logger.info("Checking if managed pod <" + podName + "> is deleted");
+    TestUtils.checkPodDeleted(podName, domainNS);
+
+    replicaCnt = TestUtils.getClusterReplicas(domainUid, clusterName, domainNS);
+    if (replicaCnt != replicas) {
+      throw new RuntimeException(
+          "FAILURE: Cluster replica doesn't match with scaled down size "
+              + replicaCnt
+              + "/"
+              + replicas);
+    }
+  }
+
+  private void scaleUpAndVarify(int incrNum) throws Exception {
+    Map<String, Object> domainMap = domain.getDomainMap();
+    String domainUid = domain.getDomainUid();
+    String domainNS = domainMap.get("namespace").toString();
+    String managedServerNameBase = domainMap.get("managedServerNameBase").toString();
+    String clusterName = domainMap.get("clusterName").toString();
+
+    int replicaCnt = TestUtils.getClusterReplicas(domainUid, clusterName, domainNS);
+    int replicas = replicaCnt + incrNum;
+    String podName = domainUid + "-" + managedServerNameBase + replicas;
+
+    logger.info("Scale domain " + domainUid + " up to " + replicas + " managed servers");
+    operator.scale(domainUid, clusterName, replicas);
+
+    varifyPodReady(replicas);
+  }
+
+  /**
+   * Verify that after bounce Operator, it reconnects to a pre-configed and newly started MS
+   *
+   * @throws Exception
+   */
+  private void varifyPodReady(int replicas) throws Exception {
+    Map<String, Object> domainMap = domain.getDomainMap();
+    String domainUid = domain.getDomainUid();
+    String domainNS = domainMap.get("namespace").toString();
+    String managedServerNameBase = domainMap.get("managedServerNameBase").toString();
+    String clusterName = domainMap.get("clusterName").toString();
+
+    String podName = domainUid + "-" + managedServerNameBase + replicas;
+
+    logger.info("Checking if managed pod <" + podName + "> is created");
+    TestUtils.checkPodCreated(podName, domainNS);
+
+    logger.info("Checking if managed server <" + podName + "> is running");
+    TestUtils.checkPodReady(podName, domainNS);
+
+    logger.info("Checking if managed service <" + podName + "> is created");
+    TestUtils.checkServiceCreated(podName, domainNS);
+
+    int replicaCnt = TestUtils.getClusterReplicas(domainUid, clusterName, domainNS);
+    if (replicaCnt != replicas) {
+      throw new RuntimeException(
+          "FAILURE: Cluster replica doesn't match with scaled up size "
+              + replicaCnt
+              + "/"
+              + replicas);
+    }
+  }
+
+  private void stopOperator() throws Exception {
+    String testMethodName = new Object() {}.getClass().getEnclosingMethod().getName();
+    logTestBegin(testMethodName);
+
+    String operNS = operator.getOperatorNamespace();
+    String cmd = "kubectl scale --replicas=0 deployment/weblogic-operator" + " -n " + operNS;
```

Comment:
add a method stopUsingReplicas in Operator.java and call that from this test.

Reply:
If you like to move     stopOperator() to Operator.java, I'd suggest we move startOperator() to Operator as well. After we scale down the Operator to 0, we want to scale it up to 1,not deploy it again, do we?

Reply:
Yes, move both. I added a comment for start as well.